src/Components/Web.JS/src/Boot.WebAssembly.ts

```diff
@@ -91,6 +91,9 @@ async function boot(options?: Partial<WebAssemblyStartOptions>): Promise<void> {
     getParameterValues: (id) => BINDING.js_string_to_mono_string(componentAttacher.getParameterValues(id) || ''),
   };
 
+  const persistedState = discoverPersistedState(document);
+  window['Blazor']._internal.getPersistedState = () => BINDING.js_string_to_mono_string(persistedState);
```

Comment:
Maybe it's not necessary, but if we wanted, we could bypass the whole .NET string representation of the data. That is, we could do the base64-decoding on the JS side, and supply a `byte[]` to .NET via the unmarshalled interop.

This could be left as a possible future optimization.

Reply:
That sounds like a good thing to do, I haven't looked at the perf just yet because it's a one time operation

Reply:
Scheduled this for a follow-up PR